### PR TITLE
fix #523: add heartbeats for front-end

### LINKED ISSUE #523: Front-end WebSocket connections are only alive for ~1min.
This is because the WebSocket server will close the connection if it does not receive any message within an expiry time (~1min).

The WebSocket server does send heartbeats, but the front-end code does not respond to them.

### PR DESCRIPTION
**Pull request by twd2**

### CODE REVIEW

File: vj4/ui/pages/home_messages.page.js [hunk 1]
```diff
@@ -36,6 +36,11 @@ const page = new NamedPage('home_messages', () => {
     reduxStore = store;
 
     const sock = new SockJs('/home/messages-conn');
+
+    setInterval(() => {
```
Comment (moesoha) [inline vj4/ui/pages/home_messages.page.js]:
```javascript
let heartbeatClock;
sock.onopen = () => { heartbeatClock = setInterval(/* heartbeat */); };
sock.onclose = () => clearInterval(heartbeatClock);
```

File: vj4/ui/pages/record_main.page.js [hunk 2]
```diff
@@ -8,6 +8,10 @@ const page = new NamedPage('record_main', async () => {
   const sock = new SockJs(Context.socketUrl);
   const dd = new DiffDOM();
 
+  setInterval(() => {
```
Comment (moesoha) [inline vj4/ui/pages/record_main.page.js]: ibid

File: vj4/ui/pages/record_detail.page.js [hunk 3]
```diff
@@ -7,6 +7,10 @@ const page = new NamedPage('record_detail', async () => {
   const sock = new SockJs(Context.socketUrl);
   const dd = new DiffDOM();
 
+  setInterval(() => {
```
Comment (moesoha) [inline vj4/ui/pages/record_detail.page.js]: ibid

File: vj4/ui/pages/problem_detail.page.js [hunk 4]
```diff
@@ -188,6 +188,11 @@ const page = new NamedPage(['problem_detail', 'contest_detail_problem', 'homewor
     } = await loadReactRedux(ScratchpadReducer);
 
     const sock = new SockJs(Context.socketUrl);
+
+    setInterval(() => {
```
Comment (moesoha) [inline vj4/ui/pages/problem_detail.page.js]: ibid

File: vj4/ui/pages/judge_playground.page.js [hunk 5]
```diff
@@ -7,6 +7,10 @@ const page = new NamedPage('judge_playground', async () => {
 
   const sock = new SockJs('/judge/consume-conn');
 
+  setInterval(() => {
```
Comment (moesoha) [inline vj4/ui/pages/judge_playground.page.js]: ibid

File: vj4/ui/pages/judge_playground.page.js [hunk 6]
```diff
@@ -114,6 +119,8 @@ const page = new NamedPage('judge_playground', async () => {
     `)
       .appendTo(div);
   };
+
+  clearInterval(heartbeatClock)
```
Comment (moesoha) [inline vj4/ui/pages/judge_playground.page.js:123]: Wrong place? should go before L121

Comment (twd2) [reply vj4/ui/pages/judge_playground.page.js:123]: gg

**Pull request merged**